### PR TITLE
feat: voyageai support

### PR DESCRIPTION
**Pull request by fzowl**
Adding VoyageAI embedding and rerank support

### CODE REVIEW

File: python/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "lancedb-python"
-version = "0.16.0-beta.0"
+version = "0.16.0-beta.1"
```
Comment (wjones127) [inline python/Cargo.toml:3]: You do not need to bump the version. That happens automatically in release process.

Comment (fzowl) [reply python/Cargo.toml]: No, i don't need - i did not find the rules, so i thought the minor versions are raised every time. I'm reverting back this change.

Comment (fzowl) [reply python/Cargo.toml]: @wjones127 Can you please check the PR again?

**Comment by AyushExel** [comment]
Hey @fzowl looks good, can you please address the following:
- Fix linting 
- we'll also need to add it to the docs pages for better discoverability - [Rerankers](https://lancedb.github.io/lancedb/reranking/) , [Embedding functions](https://lancedb.github.io/lancedb/embeddings/default_embedding_functions/). You can also do it in a follow up PR.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**